src/main/java/shop/service/impl/ActionStrategyHandlerImpl.java

```diff
@@ -0,0 +1,18 @@
+package shop.service.impl;
+
+import java.util.HashMap;
+import shop.service.action.ActionHandler;
+import shop.service.action.ActionStrategyHandler;
+
+public class ActionStrategyHandlerImpl implements ActionStrategyHandler {
+    private final HashMap<String, ActionHandler> actionStrategyHashMap;
+
+    public ActionStrategyHandlerImpl(HashMap<String, ActionHandler> actionStrategyHashMap) {
+        this.actionStrategyHashMap = actionStrategyHashMap;
+    }
```

Comment:
here everything is good
let's inject dependencies like this - through the constructor